src/decoder.rs

```diff
@@ -291,7 +292,13 @@ impl<R: Read> Decoder<R> {
         loop {
             let marker = match pending_marker.take() {
                 Some(m) => m,
-                None => self.read_marker()?,
+                None => match self.read_marker() {
+                    Err(e) => {
```

Comment:
Do we really want to ingore all IO errors? see `io::Error::kind()` and `ErrorKind::UnexpectedEof`, I would start with a small set.